sympy/geometry/util.py

```diff
@@ -696,9 +697,11 @@ def intersection(*entities, pairwise=False, **kwargs):
 
     # entities may be an immutable tuple
     entities = list(entities)
+    y = Dummy()
     for i, e in enumerate(entities):
         if not isinstance(e, GeometryEntity):
             entities[i] = Point(e)
+        entities[i] = entities[i].replace(lambda x: x.is_Float, lambda x: nsimplify(x, rational=True) + y).xreplace({y: 0})
```

Comment:
```suggestion
        entities[i] = entities[i].replace(lambda x: x.is_Float, lambda x: nsimplify(x, rational=True))
```

Reply:
I am not sure why you are doing this. What is `y` for?

Reply:
I am not entirely certain as to why, but using the dummy object `y` gives out the correct result when using floats in ellipsis. Removing `y` results in an empty intersection result from what I have tested(as if the rationalization did not take effect!).

Reply:
My guess is that this is needed so the replacement will take place since otherwise 2 would not replace 2.0